### DISCUSSION TITLE
recommended way to start ghostty maximized

### DISCUSSION THREAD

**Discussion by pdeva** [ghostty-org/ghostty#3642] [Q&A] [answered]
on mac os the best way i have found so far is to set window width and height to abnormally large size and then window always starts maximized

```
window-height=510
window-width=1640
```

wondering if there is any other recommended way?

**Comment by JakeWharton** [comment]
Maybe could be done as part of the work on #3362

**Comment by 00-kat** [comment]
From [the documentation for those configuration keys](https://ghostty.org/docs/config/reference#window-width):

> Sizes larger than the screen size will be clamped to the screen size. This can be used to create a maximized-by-default window size.

Hence I think the way you're doing it is the recommended way.

**Comment by knownasnaffy** [comment]
While attempting to start the app in a maximized state, I found that there is no direct configuration option to achieve this.

As mentioned in the documentation, I tried setting the window width and height to values larger than the screen size. This approach forces the window to clamp to the screen dimensions, but it only adjusts the size and does not change the actual window state. As a result, the window is not truly maximized and retains its rounded corners, as shown in the image below:

![image](https://github.com/user-attachments/assets/a1ad76f6-31eb-4b4f-88a2-18361dd4f486)
![image](https://github.com/user-attachments/assets/ee8f6be5-ed43-484b-9b8e-d750798cf8f2)

I am using KDE as my desktop environment on Manjaro. To resolve this, I attempted to use KDE's Window Rules feature to enforce a maximized state on startup. However, the application resisted this rule and reverted to its default dimensions a moment after it started.

A potential improvement would be to add a direct configuration option to start the application in a maximized state.

**Reply by knownasnaffy** [reply]
I also tried using the [`window-save-state`](https://ghostty.org/docs/config/reference#window-save-state) setting hoping to achieve the same result but that didn't work either

**Reply by basile-henry** [reply]
> the window is not truly maximized and retains its rounded corners

I'm also on KDE, and it's worse than that when you open ghostty in a monitor that isn't the primary one (with the taskbar) because then the ghostty window leaves space at the bottom of the screen for a taskbar that isn't there (I'm guessing it defaults to using the available size of the primary monitor instead of the current one?)

**Reply by knownasnaffy** [reply]
> (I'm guessing it defaults to using the available size of the primary monitor instead of the current one?)
Sorry mate, can't provide much feedback on that, I've got a single monitor

**Comment by asyncci** [comment]
fullscreen = true

**Reply by 00-kat** [reply]
Fullscreen isn't the same as maximised.

**Comment by jmckillican** [comment]
if you have this in your config to save window state/size and set it to full screen, when you quit the ghostty and reopen it will reopen as full screen:

`window-save-state = always`

**Comment by mariovictorrs** [comment]
Maybe an option in the config file might be ideal, similar to `fullscreen`.
I implemented this idea (if you wanna try and give you thoughts, you're welcome) in [this](https://github.com/ComicShrimp/ghostty/tree/feat/config_gtk_add_start_maximized_option) branch, and all seems to work just fine with the new flag `gtk-maximize`.

In this implementation, the `fullscreen` mode, if set, will overwrite the maximized mode

**Reply by knownasnaffy** [reply]
Looks great 😃 Care to start a PR?

**Reply by mariovictorrs** [reply]
Not a problem!, I'll make the issue and then start a PR, thanks for taking a look!

**Reply by tib-rijo** [reply]
Looking forward to this.

**Comment by chenzhiwei** [comment]
Hello, does anyone find a solution for this?

**Comment by rhodes-b** [comment, marked as answer]
https://ghostty.org/docs/config/reference#maximize

there is a config option for maximize

**Reply by 00-kat** [reply]
Added in https://github.com/ghostty-org/ghostty/commit/844f20d01f0d3482f76bd61ab2815506cc935b5c, it appears.

**Reply by chenzhiwei** [reply]
wow, great. I did not see this before.

**Reply by Diestroid** [reply]
Thanks man

**Reply by AlexSlayerLoop** [reply]
it does not works for me on macos  i'm running version 1.1.3 (up to date as for now) just fullscreen at the moment

**Reply by alexeykiselev** [reply]
For macOS, I found another combination:
```config
fullscreen=true
macos-non-native-fullscreen=visible-menu
```
It looks a bit ugly to me, no window decorations at all, but it does resemble a maximized window without borders. One upside: you can’t accidentally move the window with the mouse.

And as the documentation states: tabs do not work in this mode. But that’s not a problem for me, I don’t use them.

**Reply by tib-rijo** [reply]
Here on Linux tabs work in the present moment.

**Comment by obriat** [comment]
`maximize = true ` doesn't work (on Linux) 
@pdeva solution  with large dimensions works
